templates/vue3/src/App.vue

```diff
@@ -0,0 +1,160 @@
+                <hr />
+                <p>
+                    To keep learning, check out <a target="_blank" rel="noreferrer" href="https://docs.near.org">the NEAR docs</a> or look through some <a target="_blank" rel="noreferrer" href="https://examples.near.org">example apps</a>.
+                </p>
+            </main>
+            <Notification v-show="notificationVisible" 
+                          ref="notification"
+                          :networkId="networkId"
+                          :msg="'called method: setGreeting'"
+                          :visible="false"
+                          />
+        </template>
+
+        <template v-else>
+            <main>
+                <h1>Welcome to NEAR!</h1>
+                <p>
+                    To make use of the NEAR blockchain, you need to sign in. The button
+                    below will sign you in using NEAR Wallet.
+                </p>
+                <p>
+                    By default, when your app runs in "development" mode, it connects
+                    to a test network ("testnet") wallet. This works just like the main
+                    network ("mainnet") wallet, but the NEAR Tokens on testnet aren't
+                    convertible to other currencies - they're just for testing!
+                </p>
+                <p>
+                    Go ahead and click the button below to try it out:
+                </p>
+                <p style="text-align:center; margin-top:2.5em">
+                    <button v-on:click="login">Sign in</button>
+                </p>
+            </main>
+        </template>
+    </div>
+</template>
+
+<script>
+
+import getConfig from './config'
+const nearConfig = getConfig(process.env.NODE_ENV || 'development')
+
+console.log(`networkId:${nearConfig.networkId} CONTRACT_NAME:${nearConfig.contractName}`);
+
+import { login, logout, onSubmit } from './utils'  
+import './global.css'
+
+import Notification from './components/Notification.vue'
+
+    export default {
+        created() {
+            if (this.signedIn) {
+                this.retrieveSavedGreeting()
+            }
+        },
+        name: 'App',
+        components: {
+            Notification
+        },
+        data: function () {
+            return {
+                savedGreeting: "Hello stranger",
+                newGreeting: "Hello stranger",
+                notificationVisible: false,
+                networkId: nearConfig.networkId,
+            }
+        },
+        computed: {
+            signedIn() { return window.walletConnection.isSignedIn() },
+            accountId() { return window.accountId }
+        },
+
+        methods: {
+
+            login() {
+                console.log("calling utils.login")
+                login()
+                this.retrieveSavedGreeting()
+            },
+
+            logout: logout,
+
+            retrieveSavedGreeting()  {
+                //retrieve greeting
+                var self=this
```

Comment:
If you change the `retrievSavedGreeting` line to use an arrow function, I think you can get rid of the `var self=this`

```js
retrieveSavedGreeting: () => {
  ...

Reply:
You're right, I'll use an arrow function